src/System.Slices/src/System/SpanExtensions.cs

```diff
@@ -162,6 +165,62 @@ public static Span<char> Slice(this string str, int start, int length)
             cast[0] = value;
         }
 
+        /// <summary>
+        /// Determines whether two spans are equal by comparing the elements by using generic Equals method
+        /// </summary>
+        /// <param name="first">A span of type T to compare to second.</param>
+        /// <param name="second">A span of type T to compare to first.</param>
+        public static bool SequenceEqual<T>(this Span<T> first, Span<T> second)
+            where T : struct, IEquatable<T>
+        {
+            if (first.Length != second.Length)
+            {
+                return false;
+            }
+
+            // we can not call memcmp here because structures might have nontrivial Equals implementation
+            for (int i = 0; i < first.Length; i++)
+            {
+                if (!first.GetItemWithoutBoundariesCheck(i).Equals(second.GetItemWithoutBoundariesCheck(i)))
+                {
+                    return false;
+                }
+            }
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether two spans are structurally (byte-wise) equal by comparing the elements by using memcmp
+        /// </summary>
+        /// <param name="first">A span, of type T to compare to second.</param>
+        /// <param name="second">A span, of type U to compare to first.</param>
+        public static bool StructuralEqual<[Primitive]T, [Primitive]U>(this Span<T> first, Span<U> second)
+            where T : struct
```

Comment:
Not sure if this is the most intuitive name. I'd probably call it BlockEquals  (by vague similarity with BlockCopy)